### PR TITLE
etcd: 3.3.20 -> 3.3.21, etcd_3_4: init at 3.4.8

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

`3.3` -> `3.4` isn't an easy version bump so I'd like to add `3.4` as a first step.

https://github.com/etcd-io/etcd/blob/master/CHANGELOG-3.4.md#v340-2019-08-30

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc K8s people
@johanot @offlinehacker @saschagrunert @srhb

### CODE REVIEW

**Comment by offlinehacker** [comment]
Why do we need two versions is 3.4 incompatible with 3.4?

**Comment by zowoq** [comment]
> Why do we need two versions is 3.3 incompatible with 3.4?

For `3.4` the API defaults to v3 and has changes to how it handles crypto.

The modules and tests would seem to need (non-trivial?) changes to support `3.4`.

I'd like to make `3.4` available for other platforms (and NixOS) for now until we can migrate from `3.3` to `3.4`.

File: pkgs/servers/etcd/3.4.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  installPhase = ''
+    install -Dm755 bin/* bin/functional/cmd/* -t $out/bin
+  '';
+
+  meta = with lib; {
+    description = "Distributed reliable key-value store for the most critical data of a distributed system";
+    license = licenses.asl20;
+    homepage = "https://etcd.io/";
+    maintainers = with maintainers; [ offline ];
```
Comment (offlinehacker) [inline pkgs/servers/etcd/3.4.nix:30]: Can you add yourself as maintainer?

Comment (zowoq) [reply pkgs/servers/etcd/3.4.nix]: Added myself to both.

**Review by offlinehacker — APPROVED** [review]

**Pull request merged**